bounce/db/membership.py

```diff
@@ -36,65 +42,260 @@ def to_dict(self):
+    elif (editors_role == Roles.president.value
+          and members_role != Roles.president.value):
+        return True
+    # Admins can only delete Member memberships
+    elif (editors_role == Roles.admin.value
+          and members_role == Roles.member.value):
+        return True
+    # Return false if none of the conditions are met
+    return False
+
+
+def can_update(editors_role, members_role):
+    """
+    Determines whether user can update a membership from the database
+    Args:
+        editors_role (Role):
+            the role of the member who is updating the membership
+        members_role (Role):
+            the role of the member whose membership is being updated
+    """
+    # President can update any memberships except for other presidents
+    if (editors_role == Roles.president.value
+            and members_role != Roles.president.value):
+        return True
+    # Admins can only update members membership
+    elif (editors_role == Roles.admin.value
+          and members_role == Roles.member.value):
+        return True
+    # Return false if none of the conditions are met
+    return False
+
+
+def update(session, club_name, user_id, editors_role, members_role,
+           new_position, new_role):
+    """
+    Updates membership that asscociates the give user with the given
+    club.
+    """
+    if can_update(editors_role, members_role):
+        query = f"""
+            UPDATE memberships
+            SET role = '{new_role}', position = '{new_position}'
+            WHERE memberships.user_id = '{user_id}'
+            AND memberships.club_id =
+                (SELECT id FROM clubs WHERE name = '{club_name}')
+        """
+        session.execute(query)
+        session.commit()
+
+    else:
+        raise PermissionError("Permission denied for updating membership")
+
+
+def insert(session, club_name, user_id, editors_role, members_role, position):
     """Creates a new membership that associates the given user with the given
-    club. """
-    # For now we do nothing on conflict, but when we have roles on these
-    # memberships we need to update on conflict.
-    query = """
-        INSERT INTO memberships (user_id, club_id) VALUES (
-            :user_id,
-            (SELECT id FROM clubs WHERE name = :club_name)
-        )
-        ON CONFLICT DO NOTHING
+    club.
+
+    Args:
+        editors_role (Role):
+            the role of the member who is deleting the membership
+        members_role (Role):
+            the role of the member who's membership is being added
     """
-    session.execute(query, {'user_id': user_id, 'club_name': club_name})
-    session.commit()
+    if can_insert(editors_role, members_role):
+        query = f"""
+            INSERT INTO memberships (user_id, club_id, role, position) VALUES (
+                '{user_id}',
+                (SELECT id FROM clubs WHERE name = '{club_name}'),
+                '{members_role}',
+                '{position}'
```

Comment:
Is this branch up to date with `master`? If so, it shouldn't be using f-strings to insert variables into SQL statements (because that's a SQL injection vulnerability). You can take a look at how Kevin fixed the vulnerabilities here: https://github.com/ubclaunchpad/bounce/commit/a7359333ff7ec5cbd2066b18a6b4616878c2505e.